react/Wrapper.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC } from 'react'
+import useProduct from 'vtex.product-context/useProduct'
+import { withToast } from 'vtex.styleguide'
+
+import AddToCartButton from './AddToCartButton'
+import { mapCatalogItemToCart } from './utils'
+
+interface Props {
+  isOneClickBuy: boolean
+  available: boolean
+  disabled: boolean
+  customToastUrl: string
+  showToast: Function
+}
+
+const Wrapper: FC<Props> = ({
+  isOneClickBuy,
+  available,
+  disabled,
+  customToastUrl,
+  showToast,
+}) => {
+  const productContext: ProductContextState | undefined = useProduct()
+
+  if (!productContext) {
+    throw new Error('useProduct must be used within a ProductContextProvider')
+  }
+
+  const isEmptyContext =
+    !productContext || Object.keys(productContext).length === 0
```

Comment:
```suggestion
    Object.keys(productContext).length === 0
```

Reply:
If `productContext` is `undefined` this would result in a TypeError.
I know this should not happen since I'm already checking for `!productContext` earlier, buuuut, would not hurt to check it here too.